subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationNode.java

```diff
@@ -124,14 +125,14 @@ public InitialTransformationNode(TransformationStep transformationStep, Resolvab
         public void execute(BuildOperationExecutor buildOperationExecutor, ArtifactTransformListener transformListener) {
             this.transformedSubject = buildOperationExecutor.call(new ArtifactTransformationStepBuildOperation() {
                 @Override
-                protected TransformationSubject transform() {
+                protected Try<TransformationSubject> transform() {
                     File file;
                     try {
                         file = artifact.getFile();
                     } catch (ResolveException e) {
-                        return TransformationSubject.failure("artifact " + artifact.getId().getDisplayName(), e);
+                        return Try.failure(e);
```

Comment:
Do I understand correctly that the assumption is that this failure will never be presented directly to the user? Given that we are losing the context of the artifact being transformed, right?

So we assume that the resolution failure is shown when we do try to resolve the configuration for doing the immediate transformation and we add the context there anyway, right?